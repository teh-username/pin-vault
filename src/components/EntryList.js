import React from 'react';
import { ScrollView, View, StyleSheet, Text } from 'react-native';
import { List, ListItem } from 'react-native-elements';

const EntryList = ({ entries, onPress }) =>
  entries.length === 0 ? (
    <View style={styles.container}>
      <Text style={styles.text}>Hmm, Looks like you have no entries yet.</Text>
      <Text style={styles.text}>Hit the add button below to get started!</Text>
    </View>
  ) : (
    <ScrollView>
      <List>
        {entries.map(([id, { name }]) => (
          <ListItem
            key={id}
            title={name}
            onPress={() => {
              onPress(id);
            }}
          />
        ))}
      </List>
    </ScrollView>
  );

const styles = StyleSheet.create({
  container: {
    flex: 1,
    top: 10,
    left: 10,
  },
  text: {
    fontSize: 15,
  },
});

export default EntryList;
