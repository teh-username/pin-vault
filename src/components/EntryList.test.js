import React from 'react';
import EntryList from './EntryList';

describe('EntryList component', () => {
  it('should render as expected', () => {
    const sampleEntries = [
      ['md391', { name: 'Bank 1' }],
      ['krni8', { name: 'Bank 2' }],
      ['jr278', { name: 'Bank 3' }],
    ];
    const wrapper = shallow(
      <EntryList entries={sampleEntries} onPress={() => {}} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
