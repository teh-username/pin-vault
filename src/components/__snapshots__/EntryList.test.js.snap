// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntryList component should render as expected 1`] = `
<ScrollView>
  <List>
    <ListItem
      chevronColor="#bdc6cf"
      disabled={false}
      hideChevron={false}
      key="md391"
      leftIconUnderlayColor="white"
      onPress={[Function]}
      rightIcon={
        Object {
          "name": "chevron-right",
        }
      }
      rightTitleNumberOfLines={1}
      roundAvatar={false}
      subtitleNumberOfLines={1}
      switchButton={false}
      textInputEditable={true}
      title="Bank 1"
      titleNumberOfLines={1}
      underlayColor="white"
    />
    <ListItem
      chevronColor="#bdc6cf"
      disabled={false}
      hideChevron={false}
      key="krni8"
      leftIconUnderlayColor="white"
      onPress={[Function]}
      rightIcon={
        Object {
          "name": "chevron-right",
        }
      }
      rightTitleNumberOfLines={1}
      roundAvatar={false}
      subtitleNumberOfLines={1}
      switchButton={false}
      textInputEditable={true}
      title="Bank 2"
      titleNumberOfLines={1}
      underlayColor="white"
    />
    <ListItem
      chevronColor="#bdc6cf"
      disabled={false}
      hideChevron={false}
      key="jr278"
      leftIconUnderlayColor="white"
      onPress={[Function]}
      rightIcon={
        Object {
          "name": "chevron-right",
        }
      }
      rightTitleNumberOfLines={1}
      roundAvatar={false}
      subtitleNumberOfLines={1}
      switchButton={false}
      textInputEditable={true}
      title="Bank 3"
      titleNumberOfLines={1}
      underlayColor="white"
    />
  </List>
</ScrollView>
`;
